src/ragas/metrics/faithfulnes.py

```diff
@@ -70,65 +70,76 @@ def init_model(self: t.Self):
-        final_answer = "Final verdict for each statement in order:"
-        final_answer = final_answer.lower()
-        for i, output in enumerate(outputs):
-            output = output[0].text.lower().strip()
-            if output.find(final_answer) != -1:
-                output = output[output.find(final_answer) + len(final_answer) :]
-                score = sum(
-                    0 if "yes" in answer else 1
-                    for answer in output.strip().split(".")
-                    if answer != ""
+        with trace_as_chain_group(
+            callback_group_name, callback_manager=callbacks
+        ) as batch_group:
+            for q, a in zip(question, answer):
+                human_prompt = LONG_FORM_ANSWER_PROMPT.format(question=q, answer=a)
+                prompts.append(ChatPromptTemplate.from_messages([human_prompt]))
+
+            result = generate(prompts, self.llm, callbacks=batch_group)
+            list_statements: list[list[str]] = []
+            for output in result.generations:
+                # use only the first generation for each prompt
+                statements = output[0].text.split("\n")
+                list_statements.append(statements)
+
+            prompts = []
+            for context, statements in zip(contexts, list_statements):
+                statements_str: str = "\n".join(
+                    [f"{i+1}.{st}" for i, st in enumerate(statements)]
                 )
-                score = score / len(list_statements[i])
-            else:
-                score = max(0, output.count("verdict: no")) / len(list_statements[i])
-
-            scores.append(1 - score)
-
-        return ds.add_column(f"{self.name}", scores)  # type: ignore
+                contexts_str: str = "\n".join(context)
+                human_prompt = NLI_STATEMENTS_MESSAGE.format(
+                    context=contexts_str, statements=statements_str
+                )
+                prompts.append(ChatPromptTemplate.from_messages([human_prompt]))
+
+            result = generate(prompts, self.llm, callbacks=batch_group)
+            outputs = result.generations
+
+            scores = []
+            final_answer = "Final verdict for each statement in order:"
+            final_answer = final_answer.lower()
+            for i, output in enumerate(outputs):
+                output = output[0].text.lower().strip()
+                if output.find(final_answer) != -1:
+                    output = output[output.find(final_answer) + len(final_answer) :]
+                    score = sum(
+                        0 if "yes" in answer else 1
+                        for answer in output.strip().split(".")
+                        if answer != ""
+                    )
+                    score = score / len(list_statements[i])
+                else:
+                    score = max(0, output.count("verdict: no")) / len(list_statements[i])
+
+                scores.append(1 - score)
+
+        return scores
```

Comment:
can you make sure this part is consistent with your change also. There was a merge conflict I fixed here